### DISCUSSION TITLE
How would I update a single `container` of a Pod defined in a different module?

### DISCUSSION THREAD

**Discussion by felixscheinost** [hall/kubenix#17] [Q&A] [answered]
I am importing a `DaemonSet` from a Helm chart. As `spec.template.containers` is a list of submodules I think I can't update a container defined in a separate module, right? The same is true for basic `Pod` of course.

If the `kubernetes.api.pods` were defined manually and not imported from the schema JSON, it would probably be better for composability if `spec.containers` were `attrsOf string to submodule`, right?

So I could say e.g. `spec.containers.<name of container>.ports = [<some modification>]`.

Have you thought about this already? Could we find a way to make this happen? So that in Nix `containers` is an `attrSet` which gets converted into a flat list of values on serialization?

**Comment by felixscheinost** [comment, marked as answer]
Sorry, answered my own question 😅 

Actually it already is implemented that way! (`coerceAttrsOfSubmodulesToListByKey`)

So I could write `containers."<name>".ports = ...`.

**Comment by hall** [comment]
For future travelers, I've added a section to the docs which declares this functionality: https://kubenix.org/examples/helm/#patching
